app/src/main/java/com/tohami/newsapi/ui/news/NewsFragment.java

```diff
@@ -0,0 +1,105 @@
+package com.tohami.newsapi.ui.news;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.Nullable;
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.navigation.Navigation;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import com.tohami.newsapi.R;
+import com.tohami.newsapi.adapter.NewsAdapter;
+import com.tohami.newsapi.base.BaseFragment;
+import com.tohami.newsapi.data.model.NewsArticle;
+import com.tohami.newsapi.databinding.FragmentNewsBinding;
+import com.tohami.newsapi.di.Injection;
+
+import java.util.ArrayList;
+
+public class NewsFragment extends BaseFragment implements NewsAdapter.OnItemClickedListener {
+
+    private NewsViewModel mViewModel;
+    private FragmentNewsBinding binding;
+
+
+    public static NewsFragment newInstance() {
+        return new NewsFragment();
+    }
+
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
```

Comment:
All logic is included in a single function, this is not a good practice, move each separate functionality to its own separate function. as a general rule, a single method should perform a single function

Reply:
done